lib/chef/provisioning/docker_driver/docker_transport.rb

```diff
@@ -27,6 +27,10 @@ def execute(command, options={})
         opts[:stdin] = true
       end
 
+      if options[:read_timeout]
+        Docker.options[:read_timeout] = options[:read_timeout]
```

Comment:
I think this needs to be `opts[:wait] = `options[:read_timeout]`. Otherwise if you set the timeout on one `machine` it will affect all the others. I take this from here: https://github.com/swipely/docker-api/blob/master/lib/docker/exec.rb#L51